### PR TITLE
[FMV][AArch64] Remove features which expose non exploitable runtime behavior.

### PR DESCRIPTION
**Pull request by labrinea**
Features ebf16, memtag3, and rpres allow existing instructions to behave differently depending on the value of certain control registers. FMV does not read the content of control registers making these features unsuitable for runtime dispatch. See the ACLE patch for more info: https://github.com/ARM-software/acle/pull/355

### CODE REVIEW

File: compiler-rt/lib/builtins/cpu_model/aarch64/fmv/mrs.inc [hunk 1]
```diff
@@ -45,8 +45,6 @@ static void __init_cpu_features_constructor(unsigned long hwcap,
     setCPUFeature(FEAT_SSBS2);
   if (hwcap2 & HWCAP2_MTE)
     setCPUFeature(FEAT_MEMTAG2);
-  if (hwcap2 & HWCAP2_MTE3)
```
Comment (ilinpv) [inline compiler-rt/lib/builtins/cpu_model/aarch64/fmv/mrs.inc:48]: Do we need to remove runtime detection for these features for other targets as well?

**Review by ilinpv — APPROVED** [review]
I wonder if we can get any cases of using asm msr in function version for features in question? Anyway LGTM to be aligned with specification.

File: compiler-rt/lib/builtins/cpu_model/aarch64/fmv/mrs.inc (hunk 1 above)
Comment (labrinea) [reply compiler-rt/lib/builtins/cpu_model/aarch64/fmv/mrs.inc:48]: There are no uses of the removed features in other targets.

**Comment by labrinea** [comment]
> I wonder if we can get any cases of using asm msr in function version for features in question? Anyway LGTM to be aligned with specification.

Dit was the only FMV feature with a corresponding PSTATE value which could be dirrectly used with MRS/MSR-immediate. The rest of the features have corresponding bits in FPCR and SCTLR_ELx. Those registers can be accessed in the absence of rpres, ebf16, and memtag3, that's why we don't have a Subtarget feature for them in the LLVM backend.

**Review by jroelofs — APPROVED** [review]

**Pull request merged**